github/git_refs.go

```diff
@@ -58,13 +60,53 @@ func (s *GitService) GetRef(ctx context.Context, owner string, repo string, ref
 
 	r := new(Reference)
 	resp, err := s.client.Do(ctx, req, r)
-	if err != nil {
+	if _, ok := err.(*json.UnmarshalTypeError); ok {
+		// Multiple refs, means there wasn't an exact match.
+		return nil, resp, nil
+	} else if err != nil {
 		return nil, resp, err
 	}
 
 	return r, resp, nil
 }
 
+// GetRefs fetches a slice of Reference objects for a given Git ref.
+// If a Git ref does not match exactly, GitHub will return all refs that match it as a prefix.
+// Example:
+// feature -> [featureA, featureB]
+// featureA -> [featureA]
+//
+// GitHub API docs: https://developer.github.com/v3/git/refs/#get-a-reference
+func (s *GitService) GetRefs(ctx context.Context, owner string, repo string, ref string) ([]*Reference, *Response, error) {
+	ref = strings.TrimPrefix(ref, "refs/")
+	u := fmt.Sprintf("repos/%v/%v/git/refs/%v", owner, repo, ref)
+	req, err := s.client.NewRequest("GET", u, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var rawJSON json.RawMessage
+	resp, err := s.client.Do(ctx, req, &rawJSON)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	var rs []*Reference
+	r := new(Reference)
+	// prioritize the most common case: a single returned ref
```

Comment:
Go style suggests using proper english sentences, unless the comment is a short phrase. So capitalize the first letter and add a period to this comment. I'd also put it on top of `r := new(Reference)`, so:

```Go
// Prioritize the most common case: a single returned ref.
r := new(Reference)
singleUnmarshalErr := json.Unmarshal(rawJSON, r)
```